File: app/src/androidTest/java/de/christinecoenen/code/zapp/app/mediathek/ui/detail/MediathekPlayerActivityTest.java
package de.christinecoenen.code.zapp.app.mediathek.ui.detail;

import android.app.Activity;
import android.content.Context;
import android.content.Intent;
import android.content.pm.ActivityInfo;

import androidx.test.core.app.ActivityScenario;
import androidx.test.ext.junit.runners.AndroidJUnit4;
import androidx.test.platform.app.InstrumentationRegistry;

import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;

import de.christinecoenen.code.zapp.app.ZappApplication;
import de.christinecoenen.code.zapp.app.mediathek.MediathekData;
import de.christinecoenen.code.zapp.app.mediathek.model.PersistedMediathekShow;

@RunWith(AndroidJUnit4.class)
public class MediathekPlayerActivityTest {

	private PersistedMediathekShow persistedShow;

	@Before
	public void Setup() {
		ZappApplication zappApplication = (ZappApplication) InstrumentationRegistry
			.getInstrumentation()
			.getTargetContext()
			.getApplicationContext();

		persistedShow = zappApplication.getMediathekRepository()
			.persistOrUpdateShow(MediathekData.getTestShow())
			.blockingFirst();
	}

	@Test
	public void testRecreation() {
		Context context = InstrumentationRegistry.getInstrumentation().getTargetContext();
		Intent intent = MediathekPlayerActivity.getStartIntent(context, persistedShow.getId());

		ActivityScenario<Activity> scenario = ActivityScenario.launch(intent);
		scenario.recreate();

		scenario.onActivity(activity -> activity.setRequestedOrientation(ActivityInfo.SCREEN_ORIENTATION_LANDSCAPE));
		scenario.recreate();
	}

}

File: app/src/androidTest/java/de/christinecoenen/code/zapp/app/mediathek/ui/detail/MediathekPlayerActivityTest.kt
package de.christinecoenen.code.zapp.app.mediathek.ui.detail

import android.app.Activity
import android.content.pm.ActivityInfo
import androidx.test.core.app.ActivityScenario
import androidx.test.ext.junit.runners.AndroidJUnit4
import androidx.test.platform.app.InstrumentationRegistry
import de.christinecoenen.code.zapp.app.mediathek.MediathekData
import de.christinecoenen.code.zapp.app.mediathek.ui.detail.player.MediathekPlayerActivity
import de.christinecoenen.code.zapp.models.shows.PersistedMediathekShow
import de.christinecoenen.code.zapp.repositories.MediathekRepository
import org.junit.Before
import org.junit.Test
import org.junit.runner.RunWith
import org.koin.test.KoinTest
import org.koin.test.inject

@RunWith(AndroidJUnit4::class)
class MediathekPlayerActivityTest : KoinTest {

	private val mediathekRepository: MediathekRepository by inject()
	private lateinit var persistedShow: PersistedMediathekShow

	@Before
	fun setup() {
		persistedShow = mediathekRepository
			.persistOrUpdateShow(MediathekData.testShow)
			.blockingFirst()
	}

	@Test
	fun testRecreation() {
		val context = InstrumentationRegistry.getInstrumentation().targetContext

		val intent = MediathekPlayerActivity.getStartIntent(context, persistedShow.id)
		val scenario = ActivityScenario.launch<Activity>(intent)

		scenario.recreate()
		scenario.onActivity { it.requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_LANDSCAPE }
		scenario.recreate()
	}
}
